control-plane/csi-driver/src/bin/node/registration.rs

```diff
@@ -0,0 +1,49 @@
+use csi_driver::client::RestApiClient;
+use snafu::Snafu;
+use std::{collections::HashMap, time::Duration};
+use tokio::task::JoinError;
+use tracing::trace;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)), context(suffix(false)))]
+pub(crate) enum RegistrationError {
+    TokioTaskWait { source: JoinError },
+}
+
+/// Default registration interval.
+const REGISTRATION_INTERVAL_ON_SUCCESS: Duration = Duration::from_secs(60 * 5);
+/// Default registration interval on error.
+const REGISTRATION_INTERVAL_ON_ERROR: Duration = Duration::from_secs(30);
+
+pub(crate) async fn run_registration_loop(
+    id: String,
+    endpoint: String,
+    labels: Option<HashMap<String, String>>,
+) -> Result<(), RegistrationError> {
+    let client = RestApiClient::get_client();
+
+    tokio::spawn(async move {
+        loop {
+            let interval_duration = match client.register_app_node(&id, &endpoint, &labels).await {
+                Ok(_) => {
+                    // If register_app_node is successful, set the interval to
+                    // REGISTRATION_INTERVAL_ON_SUCCESS.
+                    REGISTRATION_INTERVAL_ON_SUCCESS
+                }
+                Err(e) => {
+                    trace!("Failed to register app node: {:?}", e);
```

Comment:
Can we have similar to the io-engine, first error register, then retry silently?